fauna/client/client.py

```diff
@@ -48,6 +51,19 @@ class QueryOptions:
   additional_headers: Optional[Dict[str, str]] = None
 
 
+@dataclass
+class StreamOptions:
+  """
+    A dataclass representing options available for a stream.
+
+    * idle_timeout - Controls the maximum amount of time the driver will wait on stream idling.
+    * retry_on_timeout - If true, streaming is reconnected on timeouts.
```

Comment:
I'm curious if these were product requirements? It doesn't feel like we need to expose these at all. Out service might close the stream if it's idle too long, and the client should assume that it should reconnect.

Another consideration is that this timeout will override the read timeout on the client, which is maybe fine, but also maybe unnecessary.

TBH, I wish I never exposed read timeout on this client.